protocol/chainlib/rest.go

```diff
@@ -389,11 +389,9 @@ func (rcp *RestChainProxy) SendNodeMsg(ctx context.Context, ch chan interface{},
 
 	res, err := httpClient.Do(req)
 	if err != nil {
-		if err != nil && connectCtx.Err() == context.DeadlineExceeded {
-			// Not an rpc error, return provider error without disclosing the endpoint address
-			return nil, "", nil, utils.LavaFormatError("Provider Failed Sending Message", context.DeadlineExceeded)
+		if handledError := rcp.BaseChainProxy.ErrorHandler.HandleRestErrors(connectCtx, err); handledError != nil {
```

Comment:
the new code has a path where it doesn't return an error, the old code doesn't it returns an error from this if case in both cases, is this intentional?